lib/graphql/schema/build_from_definition.rb

```diff
@@ -115,6 +157,12 @@ def build(document, default_resolve: DefaultResolve)
 
             directives directives.values
           end
+
+          if default_resolve.respond_to? :after_build_schema
```

Comment:
Hmm, here's an alternate approach, what do you think: 

- default_resolve objects _may_ implement `#resolve_type`, and if they do, it's hooked up to the schema: 

  ```ruby 
  Schema.define do 
    if default_resolve.respond_to?(:resolve_type)
      resolve_type ->(obj, ctx) { default_resolve.resolve_type(obj, ctx) } 
    end 
  end 
  ```
  
  Then, `ResolveMap` could implement it to fetch from the provided hash: 
  
  ```ruby
  def resolve_type(obj, ctx) 
    @resolver_hash.fetch("__resolve_type").call(obj, ctx)
  end 
  ```

  Alternatively, you could check the hash during `#initialize` and conditionally define the method with `define_singleton_method`. This would be nice because you'd get `GraphQL::Schema`'s built-in check for the presence of `resolve_type`. Either way. 

- Ok, I don't have a good idea for scalars yet ... let me think on it ...